### PR TITLE
agrep 3.41.5 (new formula)

### PR DESCRIPTION
**Pull request by asmod3us**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --new-formula <formula>` (after doing `brew install <formula>`)?

---

### CODE REVIEW

File: Formula/agrep.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  desc "Approximate GREP for fast fuzzy string searching."
+  homepage "https://www.tgries.de/agrep/"
+  url "https://github.com/Wikinaut/agrep/archive/master.tar.gz"
+  version "3.41.5"
+  sha256 "b15ffb7319d8e7ec302b45d6693479a41f97c4a4f66500d30f65388620c4e24c"
+
+  conflicts_with "tre", :because => "tre also ships an agrep binary"
+
+  patch do
+    # Fixes compilation errors on OSX but is unmerged since end of June
```
Comment (MikeMcQuaid) [inline Formula/agrep.rb]: Would like to see this at least merged before we include this.

File: Formula/agrep.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+class Agrep < Formula
+  desc "Approximate GREP for fast fuzzy string searching."
+  homepage "https://www.tgries.de/agrep/"
+  url "https://github.com/Wikinaut/agrep/archive/master.tar.gz"
```
Comment (MikeMcQuaid) [inline Formula/agrep.rb]: This needs to be a tagged, stable release.

File: Formula/agrep.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+  patch do
+    # Fixes compilation errors on OSX but is unmerged since end of June
+    url "https://github.com/Wikinaut/agrep/pull/11.patch"
+    sha256 "4fc6a0f4a624d8345106f277a0efa2e0ba2f62a5db399c4f1e5699d91aff697e"
+  end
+
+  def install
+    system "make"
+    bin.install "agrep"
```
Comment (MikeMcQuaid) [inline Formula/agrep.rb]: This should use a `make install` or one should be added/requested upstream.

File: Formula/agrep.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  end
+
+  test do
+    assert_equal "Grand Total: 1 match(es) found.\n",
+      shell_output("f=$(mktemp \"${TMPDIR:-/tmp}/test.XXXXXXXXX\"); echo $'Foo\\nBar' >> \"$f\"; #{bin}/agrep -u -s -1 'Fool' \"$f\"", 1)
```
Comment (MikeMcQuaid) [inline Formula/agrep.rb]: This should use Ruby logic rather than a bunch of Bash commands/variables.

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

File: Formula/agrep.rb (hunk 2 above)
Comment (asmod3us) [inline Formula/agrep.rb]: Upstream project does not have tagged releases but publishes version numbers.

File: Formula/agrep.rb (hunk 1 above)
Comment (asmod3us) [inline Formula/agrep.rb]: Any help welcome to expedite pull request with agrep.

File: Formula/agrep.rb (hunk 3 above)
Comment (asmod3us) [inline Formula/agrep.rb]: Upstream project would need to implement a make install.

**Comment by asmod3us** [comment]
Help welcome to get a pull request merged with the upstream project to address the requested changes. I'll leave this as is until it happens.

**Comment by MikeMcQuaid** [comment]
Passing on this until the mentioned changes make their way upstream, sorry. Please consider [creating your own tap](https://github.com/Homebrew/brew/blob/master/docs/How-to-Create-and-Maintain-a-Tap.md#how-to-create-and-maintain-a-tap) instead. Thanks!

**Pull request closed without merging**